app/models/service_template_ansible_playbook.rb

```diff
@@ -24,13 +24,42 @@ def self.default_retirement_entry_point
   #     :reconfigure (same as provision)
   #
   def self.create_catalog_item(options, _auth_user)
+    task_id = create_catalog_item_queue(options, _auth_user)
+    task = MiqTask.wait_for_taskid(task_id)
+    task.task_results
+  end
+
+  def self.create_catalog_item_queue(options, auth_user)
+    task_opts = {
+      :action => "Create Ansible Playbook Service Template",
+      :userid => "system"
+    }
+
+    # This is a lookup for now
+    playbook = ManageIQ::Providers::AnsibleTower::AutomationManager::Playbook.find(options[:config_info][:provision][:playbook_id])
+    internal_tower = playbook.configuration_script_source.manager
```

Comment:
no need to called `internal_tower` and the playbook has its own manager. how about `tower = playbook.manager`?